src/virtual-machine.js

```diff
@@ -542,17 +544,59 @@ class VirtualMachine extends EventEmitter {
     /**
      * Get an SVG string from storage.
      * @param {int} costumeIndex - the index of the costume to be got.
-     * @return {string} the costume's SVG string, or null if it's not an SVG costume.
+     * @return {string|HTMLImageElement} the costume's SVG string if it's SVG,
+     *     an HTMLImageElement if it's a PNG, or null if it couldn't be found or decoded.
      */
-    getCostumeSvg (costumeIndex) {
+    getCostume (costumeIndex) {
         const id = this.editingTarget.getCostumes()[costumeIndex].assetId;
-        if (id && this.runtime && this.runtime.storage &&
-                this.runtime.storage.get(id).dataFormat === 'svg') {
+        if (!id || !this.runtime || !this.runtime.storage) return null;
+        const format = this.runtime.storage.get(id).dataFormat;
+        if (format === this.runtime.storage.DataFormat.SVG) {
             return this.runtime.storage.get(id).decodeText();
+        } else if (format === this.runtime.storage.DataFormat.PNG) {
```

Comment:
The image can also be a .jpg, so you can just have an `else` here, and then not override the MIME type in your call to `storage.get` below.

Reply:
I'm not sure what you mean by overriding a mime type?

Reply:
Sounds good, maybe just add an || then and take out the optional argument in the storage function call.